internal/plugin/process_monitor.go

```diff
@@ -11,31 +11,25 @@ import (
 	"time"
 
 	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/config"
 	"github.com/nginx/agent/v3/internal/datasource/host"
 	"github.com/nginx/agent/v3/internal/model"
 )
 
 type GetProcessesFunc func() ([]*model.Process, error)
 
-type ProcessMonitorParameters struct {
-	MonitoringFrequency time.Duration
-	getProcessesFunc    GetProcessesFunc
-}
-
 type ProcessMonitor struct {
-	params      *ProcessMonitorParameters
-	processes   []*model.Process
-	messagePipe bus.MessagePipeInterface
+	monitoringFrequency time.Duration
+	processes           []*model.Process
+	messagePipe         bus.MessagePipeInterface
+	getProcessesFunc    GetProcessesFunc
 }
 
-func NewProcessMonitor(params *ProcessMonitorParameters) *ProcessMonitor {
-	if params.getProcessesFunc == nil {
-		params.getProcessesFunc = host.GetProcesses
-	}
-
+func NewProcessMonitor(agentConfig *config.Config) *ProcessMonitor {
 	return &ProcessMonitor{
-		params:    params,
-		processes: []*model.Process{},
+		monitoringFrequency: agentConfig.ProcessMonitor.MonitoringFrequency,
+		processes:           []*model.Process{},
+		getProcessesFunc:    host.GetProcesses,
 	}
 }
```

Comment:
Extremely off-topic, feel free to ignore:

Going back to the "Updating a config endpoint" scenario that we discussed at the office, would it not be better if the plugin just saved a pointer to `agentConfig` and always read from that? That way the underlying `config.Config` could be updated by our "Config update service" we have and we could just use the same config pointer here.

Reply:
We will still have to notify plugins and services of the config changes but lets discuss this again when we start looking at that work.